Svc/OsTime/OsTime.cpp

```diff
@@ -0,0 +1,58 @@
+// ======================================================================
+// \title  OsTime.cpp
+// \author kubiak
+// \brief  cpp file for OsTime component implementation class
+// ======================================================================
+
+#include "Svc/OsTime/OsTime.hpp"
+#include "FpConfig.hpp"
+
+#include <Fw/Time/TimeInterval.hpp>
+
+namespace Svc {
+
+// ----------------------------------------------------------------------
+// Component construction and destruction
+// ----------------------------------------------------------------------
+
+OsTime ::OsTime(const char* const compName) :
+    OsTimeComponentBase(compName),
+    m_epoch_fw_time(Fw::ZERO_TIME),
+    m_epoch_os_time(),
+    m_epoch_valid(false) {}
+
+OsTime ::~OsTime() {}
+
+void OsTime::set_epoch(const Fw::Time& fw_time, const Os::RawTime& os_time) {
+    m_epoch_fw_time = fw_time;
```

Comment:
```suggestion
    (void) m_epoch_fw_time = fw_time;
```
To appease CI warnings, and in the other uses of `time_1 = time_2` below

Reply:
That approach gives me compiler errors on my machine
```
[ 77%] Building CXX object F-Prime/Svc/OsTime/CMakeFiles/Svc_OsTime.dir/OsTime.cpp.o
/home/kubiak/source/fprime/fprime-kubiak/Svc/OsTime/OsTime.cpp: In member function ‘void Svc::OsTime::set_epoch(const Fw::Time&, const Os::RawTime&)’:
/home/kubiak/source/fprime/fprime-kubiak/Svc/OsTime/OsTime.cpp:31:30: error: invalid use of ‘void’
   31 |     (void) m_epoch_fw_time = fw_time;
      |                              ^~~~~~~
/home/kubiak/source/fprime/fprime-kubiak/Svc/OsTime/OsTime.cpp:32:30: error: invalid use of ‘void’
   32 |     (void) m_epoch_os_time = os_time;
      |                              ^~~~~~~
/home/kubiak/source/fprime/fprime-kubiak/Svc/OsTime/OsTime.cpp:33:28: error: invalid use of ‘void’
   33 |     (void) m_epoch_valid = true;
      |                            ^~~~
/home/kubiak/source/fprime/fprime-kubiak/Svc/OsTime/OsTime.cpp: In member function ‘virtual void Svc::OsTime::timeGetPort_handler(FwIndexType, Fw::Time&)’:
/home/kubiak/source/fprime/fprime-kubiak/Svc/OsTime/OsTime.cpp:60:19: error: invalid use of ‘void’
   60 |     (void) time = m_epoch_fw_time;
```

Reply:
My bad sorry, looks like the syntax should be the following for return values of the assignment operator
```
    (void) (m_epoch_fw_time = fw_time);
```
That's pretty ugly 😮‍💨 

`m_epoch_valid = true;` probably doesn't need a cast

Reply:
We can also just dismiss the warning

Reply:
I'd prefer to dismiss the warnings. I'm not really sure what it even means to check the return type of an assignment outside of using the assigned variable

Reply:
Sorry, just noticing this.  Yes, dismiss all "operator=" does not check the return type.  This check is from C and not C++ compliant.  We will fix eventually.